test/YieldDirector.test.js

```diff
@@ -0,0 +1,501 @@
+        await dai.approve(treasury.address, LARGE_APPROVAL);
+
+        // Needed to spend deployer's OHM
+        await ohm.approve(staking.address, LARGE_APPROVAL);
+
+        // To get past OHM contract guards
+        await auth.pushVault(treasury.address, true)
+
+        // Initialization for sOHM contract.
+        // Set index to 10
+        await sOhm.setIndex("10000000000");
+        await sOhm.setgOHM(gOhm.address);
+        await sOhm.initialize(staking.address, treasury.address);
+
+        // Set distributor staking contract
+        await staking.setDistributor(distributor.address);
+
+        // queue and toggle reward manager
+        await treasury.queueTimelock('8', distributor.address, ZERO_ADDRESS);
+        await treasury.execute('0');
+        // queue and toggle deployer reserve depositor
+        await treasury.queueTimelock('0', deployer.address, ZERO_ADDRESS);
+        await treasury.execute('1');
+        // queue and toggle liquidity depositor
+        await treasury.queueTimelock('4', deployer.address, ZERO_ADDRESS);
+        await treasury.execute('2');
+        // queue and toggle DAI as reserve token
+        await treasury.queueTimelock('2', dai.address, ZERO_ADDRESS);
+        await treasury.execute('3');
+
+        // Deposit 10,000 DAI to treasury, 1,000 OHM gets minted to deployer with 9000 as excess reserves (ready to be minted)
+        await treasury.connect(deployer).deposit('10000000000000000000000', dai.address, '9000000000000');
+
+        // Add staking as recipient of distributor with a test reward rate
+        await distributor.addRecipient(staking.address, initialRewardRate);
+
+        // Get sOHM in deployer wallet
+        const sohmAmount = "1000000000000"
+        await ohm.approve(staking.address, sohmAmount);
+        await staking.stake(deployer.address, sohmAmount, true, true);
+
+        // Transfer 100 sOHM to alice for testing
+        await sOhm.transfer(alice.address, "100000000000");
+
+        // Approve sOHM to be deposited to Tyche
+        await sOhm.approve(tyche.address, LARGE_APPROVAL);
+        await sOhm.connect(alice).approve(tyche.address, LARGE_APPROVAL);
+    });
+
+    it('should rebase properly', async () => {
+        await expect(await sOhm.index()).is.equal("10000000000");
+
+        await triggerRebase();
+        await expect(await sOhm.index()).is.equal("10010000000");
+
+        await triggerRebase();
+        await expect(await sOhm.index()).is.equal("10020010000");
+
+        await triggerRebase();
+        await expect(await sOhm.index()).is.equal("10030030010");
+    });
+
+    it('should set token addresses correctly', async () => {
+        await tyche.deployed();
+
+        expect(await tyche.sOHM()).to.equal(sOhm.address);
+    });
+
+    it('should deposit tokens to recipient correctly', async () => {
+        // Deposit 100 sOHM into Tyche and donate to Bob
+        const principal = "100000000000";
```

Comment:
think you should use ethers.utils.parseunits('100', 9) or write a function called toSOhmDecimals() that calls ethers.utils.parseunits('100', 9). Less mistakes not having to eyeball/count the number of zeros.